jax/experimental/tf_bridge/jax_to_tf.py

```diff
@@ -0,0 +1,841 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Experimental module transforms JAX functions to be executed by TensorFlow."""
+
+import contextlib
+import functools
+import string
+import threading
+from typing import Any, Sequence
+
+import jax
+from jax import abstract_arrays
+from jax import ad_util
+from jax import core
+from jax import lax
+from jax import linear_util as lu
+from jax import numpy as jnp
+from jax import tree_util
+from jax import util
+from jax.api_util import flatten_fun
+from jax.interpreters import partial_eval as pe
+from jax.interpreters import xla
+
+import numpy as np
+import tensorflow as tf  # type: ignore[import]
+from typing import Any, Callable, Dict, Sequence
+
+# These don't have public equivalents.
+# pylint: disable=g-direct-tensorflow-import
+from tensorflow.compiler.tf2xla.python import xla as tfxla  # type: ignore[import]
```

Comment:
I believe these won't work in TensorFlow as built in opensource, because they aren't part of TF's public API, even though the code is part of TensorFlow. Or has this changed?

Reply:
These do work for me in OSS. They work even with the public 2.2.0 release.